### PR TITLE
Add toolbar clock (and game runtime) display

### LINKED ISSUE #6436: Add a system clock to the main game interface
**Describe the new feature:** 

It would be nice to have a small game wide system clock as an overlay that could sit on one of the corners of the screen. Most likely should be an option to have it visible as a setting. Makes it more convenient to get the time instead of tabbing out from the game when in exclusive full screen mode.

### PR DESCRIPTION
**Pull request by peppy**
Closes #6436

https://user-images.githubusercontent.com/191335/160235191-15bb774d-2f9d-4a79-85d2-955230d5bac8.mp4

### CODE REVIEW

File: osu.Game/Overlays/Toolbar/ToolbarClock.cs [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+                    clockDisplayMode.Value = ToolbarClockDisplayMode.Digital;
+                    break;
+
+                case ToolbarClockDisplayMode.Full:
+                    clockDisplayMode.Value = ToolbarClockDisplayMode.DigitalWithRuntime;
+                    break;
+            }
+        }
+
+        private class DigitalDisplay : ClockDisplay
```
Comment (frenzibyte) [inline osu.Game/Overlays/Toolbar/ToolbarClock.cs:108]: nit: I personally find it better to have `DigitalDisplay` and `AnalogDisplay` both stored in their own files as partials of `ToolbarClock` (i.e. `ToolbarClock_DigitalDisplay.cs` and `ToolbarClock_AnalogDisplay.cs`), as the number of private classes stored in here are somewhat obscuring the logic behind how this is working.

File: osu.Game/Overlays/Toolbar/ToolbarClock.cs [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+            {
+                const float duration = 320;
+
+                float rotation = fraction * 360 - 90;
+
+                if (Math.Abs(hand.Rotation - rotation) > 180)
+                    hand.RotateTo(rotation);
+                else
+                    hand.RotateTo(rotation, duration, Easing.OutElastic);
+            }
```
Comment (frenzibyte) [inline osu.Game/Overlays/Toolbar/ToolbarClock.cs:300]: nit: Would be best to keep private nested classes at the bottom and move this above them. (I had a hard time trying to find how this works via diff until I checked via IDE)

**Review by frenzibyte — COMMENTED** [review]
Design looks pretty good. Left a few nits on a quick pass, works well otherwise.

Was going to suggest using AM/PM depending on the user's system format, but something like `1:23:45 PM` felt quite off.

File: osu.Game/Overlays/Toolbar/ToolbarClock.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Overlays/Toolbar/ToolbarClock.cs]: I've moved them to their own files

**Review by frenzibyte — APPROVED** [review]
Pushed one small change to remove redundant/outdated `Origin` setter line. LGTM.

**Review by bdach — APPROVED** [review]

**Review by Joehuu — APPROVED** [review]

**Pull request merged**